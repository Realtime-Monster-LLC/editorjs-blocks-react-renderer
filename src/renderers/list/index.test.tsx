import React from 'react';
import { create } from 'react-test-renderer';
import List, { ListBlockData } from '.';

describe('<List />', () => {
  describe('when receives a list unordered block', () => {
    const data: ListBlockData = {
      style: 'unordered',
      items: [
        'It is a block-styled editor',
        'It returns clean data output in JSON',
        'Designed to be extendable and pluggable with a simple API',
      ],
    };

    it('renders a <ul> block', () => {
      expect(create(<List data={data} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('when receives a list ordered block', () => {
    const data: ListBlockData = {
      style: 'ordered',
      items: [
        'It is a block-styled editor',
        'It returns clean data output in JSON',
        'Designed to be extendable and pluggable with a simple API',
      ],
    };

    it('renders a <ol> block', () => {
      expect(create(<List data={data} />).toJSON()).toMatchSnapshot();
    });
  });
});
