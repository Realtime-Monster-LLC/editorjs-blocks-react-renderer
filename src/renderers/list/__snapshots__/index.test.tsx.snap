// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> when receives a list ordered block renders a <ol> block 1`] = `
<ol>
  <li>
    It is a block-styled editor
  </li>
  <li>
    It returns clean data output in JSON
  </li>
  <li>
    Designed to be extendable and pluggable with a simple API
  </li>
</ol>
`;

exports[`<List /> when receives a list unordered block renders a <ul> block 1`] = `
<ul>
  <li>
    It is a block-styled editor
  </li>
  <li>
    It returns clean 
    <b>
      data output
    </b>
     in JSON
  </li>
  <li>
    Designed to be extendable and pluggable with a simple API
  </li>
</ul>
`;
